modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/SystemIndexMigrationIT.java

```diff
@@ -64,13 +64,13 @@ protected boolean forbidPrivateIndexSettings() {
         return plugins;
     }
 
-    @Before
-    public void init() {
-        // master node is created in AbstractFeatureMigrationIntegTest#setupTestPlugin when trying to access plugins
+    @Override
+    protected PluginsService getPluginService() {
         internalCluster().setBootstrapMasterNodeIndex(0);
```

Comment:
It seems odd that asking for the PluginService is mutating the cluster. Why can't this master only node be started in a `@Before`, separately from trying to get the PluginService?

Reply:
to start a master node manually it affects other tests in this package, as they also will have to set `autoManageMasterNodes` to false
a lot of methods in InternalTestCluster are mutating (starting nodes) the cluster.
 For instance internalCluster().getCurrentMasterNodeInstance(PluginsService.class) will start a master node (default roles) if non is started yet. 
Same as internalCluster().client()

Reply:
I made all tests mange master node in migration package - I feel it should look better